cuda/src/utils/cuda_wrappers.cpp

```diff
@@ -23,4 +26,10 @@ int get_device() {
     return d;
 }
 
+cudaStream_t get_stream(const stream_wrapper& stream) {
+
+    assert(stream.stream() != nullptr);
+    return reinterpret_cast<cudaStream_t>(stream.stream());
```

Comment:
Is this safe? Surely these types are not layout-compatible? :fearful:

Reply:
I think it's "reasonably safe". Remember, `cudaStream_t` is a pointer type.

```
[bash][Legolas]:~ > grep -r "typedef.*cudaStream_t" software/cuda/11.4.3/x86_64-ubuntu2004/include/driver_types.h 
typedef __device_builtin__ struct CUstream_st *cudaStream_t;
[bash][Legolas]:~ >
```

Casting that back and forth to `void*` is not the most elegant code, but it's pretty safe.